src/components/PostDetails.tsx

```diff
@@ -98,21 +62,21 @@ export const PostDetails: React.FC<Props> = ({ post }) => {
       </div>
 
       <div className="block">
-        {!loaded && <Loader />}
+        {!loading && <Loader />}
 
-        {loaded && hasError && (
+        {loading && err && (
           <div className="notification is-danger" data-cy="CommentsError">
             Something went wrong
           </div>
         )}
 
-        {loaded && !hasError && comments.length === 0 && (
+        {loading && !err && comments.length === 0 && (
```

Comment:
The condition `loading && !err && comments.length === 0` might not be the best way to check for 'No comments yet'. It assumes that loading is still true, which might not be the case when checking for comments. Consider checking this condition after loading is complete, i.e., `!loading && !err && comments.length === 0`.